sdk/servicebus/azure-servicebus/azure/servicebus/management/_utils.py

```diff
@@ -326,14 +321,18 @@ def _validate_topic_subscription_and_rule_types(
         )
 
 def create_properties_from_dict_if_needed(properties, sb_resource_type):
-    # type: (DictPropertiesType, type) -> DictPropertiesReturnType
+    # type: (Union[PropertiesType, Mapping[str, Any]], Type[PropertiesType]) -> PropertiesType
     """
     This method is used to create a properties object given the
     resource properties type and its corresponding dict representation.
     :param properties: A properties object or its dict representation.
-    :type properties: DictPropertiesType
+    :type properties: Mapping or PropertiesType
     :param type sb_resource_type: The type of properties object.
-    :rtype: DictPropertiesReturnType
+    :rtype: PropertiesType
     """
-    return_properties = sb_resource_type(**properties) if isinstance(properties, dict) else properties
-    return return_properties
+    if isinstance(properties, sb_resource_type):
+        return properties
+    try:
+        return sb_resource_type(**cast(Mapping[str, Any], properties))
+    except TypeError:
```

Comment:
dumb question: where is this TypeError getting raised from, the `cast`?

Reply:
No, attempting to splat an object (`properties`) that is not a Mapping results in a TypeError :)
Do you think it would aid readability to move the `cast` to a separate line?

Reply:
I'm good with it :) after trying the code I understand how it works, thanks for the explanation!